src/org/omegat/gui/main/ProjectUICommands.java

```diff
@@ -537,19 +551,49 @@ protected Void doInBackground() throws Exception {
                     }
 
                     final ProjectProperties propsP = props;
-                    final File restoreOnFailFinal = restoreOnFail;
+                    final File rewriteOnSuccessFinal = rewriteOnSuccess;
+                    final boolean needToSavePropertiesFinal = needToSaveProperties;
+                    // We have to extract the processing related to needToSaveProperties{,Final}
+                    // out of  the code block of  Core.executeExclusively()
+                    // and place it here, so that we can change the value of needToSaveProperties
```

Comment:
No need to explain the modification in source comment. You should explain what and why it process like this.